### ISSUE TITLE
Pascal voc2007 problem

### ISSUE THREAD

**Issue by yulei1234** [zylo117/Yet-Another-EfficientDet-Pytorch#21]
Thanks for your good job! I have a problem. Training d0 on the pascal voc2007 dataset, the test result is almost 0, is this just the reason for the anchor ratio? Why is this result file so large...
![image](https://user-images.githubusercontent.com/37969091/79037029-5cfab580-7c00-11ea-91af-22105ccf9d90.png)

**Comment by zylo117** [comment]
Can you run efficientdet_test with your model and share the result image and a snippet of eval result file?

**Comment by yulei1234** [comment]
> 您可以与模型一起运行efficiencydet_test并共享结果图像和评估结果文件的摘要吗？

![image](https://user-images.githubusercontent.com/37969091/79037628-8ff37800-7c05-11ea-9520-14a7ee9b9b4d.png)
![image](https://user-images.githubusercontent.com/37969091/79037635-941f9580-7c05-11ea-9a66-025d342c1a08.png)

**Comment by yulei1234** [comment]
> 您可以与模型一起运行efficiencydet_test并共享结果图像和评估结果文件的摘要吗？

But my loss is very low...Can you give some suggestions

**Comment by zylo117** [comment]
> > 您可以与模型一起运行efficiencydet_test并共享结果图像和评估结果文件的摘要吗？
> 
> But my loss is very low...Can you give some suggestions

If your loss is low but the mAP is bad, it should be the anchors issus. Did you change anchors config?

**Comment by yulei1234** [comment]
> > > 您可以与模型一起运行efficiencydet_test并共享结果图像和评估结果文件的摘要吗？
> > 
> > 
> > 但是我的损失很低...你能给点建议吗
> 
> 如果您的损失很小，但mAP不好，那应该是主播问题。您是否更改了锚点配置？

I ’ll try to change the anchor. I used to think that the pascal voc and coco ’s anchor are not much different.

**Comment by zylo117** [comment]
> > > > 您可以与模型一起运行efficiencydet_test并共享结果图像和评估结果文件的摘要吗？
> > > 
> > > 
> > > 但是我的损失很低...你能给点建议吗
> > 
> > 
> > 如果您的损失很小，但mAP不好，那应该是主播问题。您是否更改了锚点配置？
> 
> I ’ll try to change the anchor. I used to think that the pascal voc and coco ’s anchor are not much different.

I don't think the anchor is the problem if you didn't change it. Could it be your groundtruth being mislabeled? Can you visualize the groundtruth on your training set?

**Comment by Guzaiwang** [comment]
In VOC label， bbox is labeled as [Xmin, Ymin, Xmax, Ymax], rather than [Xmin Ymin width height] in coco labels

**Comment by yulei1234** [comment]
> > > > > 您可以与模型一起运行efficiencydet_test并共享结果图像和评估结果文件的摘要吗？
> > > > 
> > > > 
> > > > 但是我的损失很低...你能给点建议吗
> > > 
> > > 
> > > 如果您的损失很小，但mAP不好，那应该是主播问题。您是否更改了锚点配置？
> > 
> > 
> > I ’ll try to change the anchor. I used to think that the pascal voc and coco ’s anchor are not much different.
> 
> I don't think the anchor is the problem if you didn't change it. Could it be your groundtruth being mislabeled? Can you visualize the groundtruth on your training set?
> 
> AFAIK, VOC is labeled in VOC-style annotation, not the coco-style, so it must be converted beforehand. The convertion itself might be unsuccessful.

Sorry, I was anxious to do the experiment yesterday and forgot to reply the message. There is no problem with the label conversion. I found that I made the wrong order of the category names of the voc dataset in the yml file. The problem is that after I corrected it, the result was not ideal，0.017，I am looking for something wrong

**Comment by zylo117** [comment]
> > > > > > 您可以与模型一起运行efficiencydet_test并共享结果图像和评估结果文件的摘要吗？
> > > > > 
> > > > > 
> > > > > 但是我的损失很低...你能给点建议吗
> > > > 
> > > > 
> > > > 如果您的损失很小，但mAP不好，那应该是主播问题。您是否更改了锚点配置？
> > > 
> > > 
> > > I ’ll try to change the anchor. I used to think that the pascal voc and coco ’s anchor are not much different.
> > 
> > 
> > I don't think the anchor is the problem if you didn't change it. Could it be your groundtruth being mislabeled? Can you visualize the groundtruth on your training set?
> > AFAIK, VOC is labeled in VOC-style annotation, not the coco-style, so it must be converted beforehand. The convertion itself might be unsuccessful.
> 
> Sorry, I was anxious to do the experiment yesterday and forgot to reply the message. There is no problem with the label conversion. I found that I made the wrong order of the category names of the voc dataset in the yml file. The problem is that after I corrected it, the result was not ideal，0.017，I am looking for something wrong

Can you share the tensorboard loss graph? It there any training loss/val loss comparsion?

**Comment by yulei1234** [comment]
> > > > > > > 您可以与模型一起运行efficiencydet_test并共享结果图像和评估结果文件的摘要吗？
> > > > > > 
> > > > > > 
> > > > > > 但是我的损失很低...你能给点建议吗
> > > > > 
> > > > > 
> > > > > 如果您的损失很小，但mAP不好，那应该是主播问题。您是否更改了锚点配置？
> > > > 
> > > > 
> > > > I ’ll try to change the anchor. I used to think that the pascal voc and coco ’s anchor are not much different.
> > > 
> > > 
> > > I don't think the anchor is the problem if you didn't change it. Could it be your groundtruth being mislabeled? Can you visualize the groundtruth on your training set?
> > > AFAIK, VOC is labeled in VOC-style annotation, not the coco-style, so it must be converted beforehand. The convertion itself might be unsuccessful.
> > 
> > 
> > Sorry, I was anxious to do the experiment yesterday and forgot to reply the message. There is no problem with the label conversion. I found that I made the wrong order of the category names of the voc dataset in the yml file. The problem is that after I corrected it, the result was not ideal，0.017，I am looking for something wrong
> 
> Can you share the tensorboard loss graph? It there any training loss/val loss comparsion?


I think they are normal
![image](https://user-images.githubusercontent.com/37969091/79064346-b6371780-7cda-11ea-8e9d-44e404761ee0.png)
![image](https://user-images.githubusercontent.com/37969091/79064349-bafbcb80-7cda-11ea-8cba-63e4082245af.png)

**Comment by yulei1234** [comment]
I loaded the pre-trained model, but it did not freeze

**Comment by zylo117** [comment]
you are showing the training loss, I can't see the val loss. It there a log or something?
It makes sense if you are suffering overfitting

**Comment by zylo117** [comment]
sorry for the training troubles, there's a bug in loss function. please pull the latest code and give it a try.

**Comment by yulei1234** [comment]
> sorry for the training troubles, there's a bug in loss function. please pull the latest code and give it a try.

ok,  thanks for your reminder
